sklearn/metrics/_plot/base.py

```diff
@@ -111,7 +113,7 @@ def _get_response(X, estimator, response_method, pos_label=None):
             pos_label = estimator.classes_[1]
             y_pred = y_pred[:, 1]
         else:
-            class_idx = np.flatnonzero(estimator.classes_ == pos_label)
+            class_idx = np.flatnonzero(estimator.classes_ == pos_label)[0]
             y_pred = y_pred[:, class_idx]
     else:
         if pos_label is None:
```

Comment:
If we have to convert `classes_` into a list, we can directly use the `index` method the list to make the code more readable, see the following untested code that feels more easier to follow:

```python
if pos_label is not None:
    try:
        class_idx = estimator.classes_.tolist().index(pos_label)
    except ValueError as e:
        raise ValueError(
            "The class provided by 'pos_label' is unknown. Got "
            f"{pos_label} instead of one of {estimator.classes_}"
        ) from e
else:
    class_idx = estimator.classes_[1]

...
```
and then do the handling of the `y_pred.ndim != 1` case to extract a binary `y_pred`.

Reply:
Thanks, indeed it is more readable.